framework/filters/AccessRule.php

```diff
@@ -172,7 +174,18 @@ protected function matchAction($action)
      */
     protected function matchController($controller)
     {
-        return empty($this->controllers) || in_array($controller->uniqueId, $this->controllers, true);
+        if (empty($this->controllers)) {
+            return true;
+        }
+
+        $id = $controller->getUniqueId();
+        foreach ($this->controllers as $pattern) {
+            if (fnmatch($pattern, $id)) {
```

Comment:
Why not preg_match ?

> fnmatch - For now, this function is not available on non-POSIX compliant systems except Windows. 

> using fnmatch() instead of preg_match() for frontend search expression input may be way more convenient for non-programming users.

In any case, the use of this function is alarming.